var/spack/repos/builtin/packages/cdo/package.py

```diff
@@ -32,15 +32,67 @@ class Cdo(Package):
     homepage = "https://code.zmaw.de/projects/cdo"
     url      = "https://code.zmaw.de/attachments/download/10198/cdo-1.6.9.tar.gz"
 
+    version('1.7.2', 'f08e4ce8739a4f2b63fc81a24db3ee31', url='https://code.zmaw.de/attachments/download/12760/cdo-1.7.2.tar.gz')
     version('1.6.9', 'bf0997bf20e812f35e10188a930e24e2')
 
-    variant('mpi', default=True)
+    variant('szip', default=True, description='Enable szip compression for GRIB1')
+    variant('hdf5', default=False, description='Enable HDF5 support')
+    variant('netcdf', default=True, description='Enable NetCDF support')
+    variant('udunits2', default=True, description='Enable UDUNITS2 support')
+    variant('grib_api', default=True, description='Enable GRIB_API support')
+    variant('libxml2', default=True, description='Enable libxml2 support')
+    variant('proj', default=True, description='Enable PROJ library for cartographic projections')
+    variant('curl', default=True, description='Enable curl support')
+    variant('fftw', default=True, description='Enable support for fftw3')
+    variant('magics', default=True, description='Enable Magics library support')
 
-    depends_on('netcdf')
-    depends_on('netcdf+mpi', when='+mpi')
-    depends_on('netcdf~mpi', when='~mpi')
+    depends_on('szip', when='+szip')
+    depends_on('netcdf', when='+netcdf')
+    depends_on('hdf5+threadsafe', when='+hdf5')
+    depends_on('udunits2', when='+udunits2')
+    depends_on('grib_api', when='+grib_api')
+    depends_on('libxml2', when='+libxml2')
+    depends_on('proj', when='+proj')
+    depends_on('curl', when='+curl')
+    depends_on('fftw', when='+fftw')
+    depends_on('magics', when='+magics') 
 
     def install(self, spec, prefix):
-        configure('--prefix={0}'.format(prefix))
+        config_args = ["--prefix=%s" % prefix,
```

Comment:
There isn't really any convention at the moment. The Python developers have been slowly moving from `'%s' % var` to `'{0}'.format(var)`. The former is used in Python 2 documentation while the latter is used in Python 3 documentation. The former syntax was added to Python 2.6 and 2.7 so that Python 3 scripts would be backwards compatible, but the Python 2 syntax still works in Python 3. Honestly, there's no reason we couldn't use `'--prefix=' + prefix` since it's just string concatenation.